### ISSUE TITLE
Undo typing doesn't trigger live-reload

### ISSUE THREAD

**Issue by gandalfar** [ember-cli/ember-cli#6679]
It seems that undoing typing in text editor doesn't trigger live reload on errors.

Steps to reproduce:

1. `ember s`
2. In the template:  `{{var}}`
3. (live reload happens)
4. Change template to: `{{var}}}` (triple `}}}`)
5. (live reload happens with build error)
6. Change template back to `{{var}}`
_7. live reload fails to happen_ (also app.js isn't written do `dist/assets/app.js`)
8. Now make a bigger change: `{{var}} test`
9. (live reload happens again)

I'm reading through source code of ember-cli, but I'm still tracking where in the pipeline error happens.
---

Output from `ember version --verbose && npm --version`:
```
➜  simplesass git:(master) ✗ ember version --verbose && npm --version
ember-cli: 2.11.0-beta.1-master-13de36657d
http_parser: 2.7.0
node: 6.6.0
v8: 5.1.281.83
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 57.1
modules: 48
openssl: 1.0.2h
os: darwin x64
3.10.7

```

**Comment by gandalfar** [comment]
Investigating a bit further, it doesn't seem connected to live-reload (I can reproduce it with `ember s --live-reload=false`). So the problem seems that while `ember-cli` console output says:

```
file changed templates/index.hbs

Build successful - 88ms.
```

The build of `dist/assets/app.js` doesn't happen (no changed timestamp on hard drive)

**Comment by gandalfar** [comment]
Looking at this further, it's correct that `dist/assets/app.js` doesn't get touched as it was correctly compiled in a previous step 3 (before build error). So live reload should pick up on it.

**Comment by gandalfar** [comment]
I think the problem is that livereload-server gets empty list of files that changed. So while didChange event triggers, there's no information on what actually changed. Nothing changed because `broccoli-persistent-filter` did a good job and just said that everything is cached.

I think `broccoli-persistent-filter` should invalidate cache key for `triggerError` on the file that triggered it.

**Comment by gandalfar** [comment]
My initial propose for a fix: https://github.com/gandalfar/ember-cli/commit/aa466972619b76bf95f268e3333e776b718b8bfc

**Comment by gandalfar** [comment]
It also seems to fix #6242
